## 문제
- 주어진 데이터(data/toothgrowth.csv)에는 기니피그 60마리의 치아 길이, 투여한 비타민 C 종류, 투여량이 저장되어 있다.  오렌지주스를 투여받은 기니피그의 치아 길이 평균이 아스코르브산을 투여받은 기니피그의 치아 길이 평균과 일치하는지 t-검정(Two Sample t-test)를 통해 답하고자 한다. 가설은 아래와 같다.
- 가설검정
    + 귀무가설 : 투여한 약제별 치아의 평균 길이는 같다.
    + 대립가설 : 투여한 약제별 치아의 평균 길이는 같지 않다.
- 데이터의 변수
    + len : 치아의 길이
    + supp : 투여한 약제(VC: 아스코르브산, OJ: 오렌지주스)
    + does : 투여량
- 단, 데이터의 각 변수들은 정규분포를 만족하며, 두 그룹은 등분산을 띤다.
- 문제 1: 표본평균 $\overline{X_{OJ}}$, $\overline{X_{VC}}$를 구하시오.
- 문제 2: 위의 가설을 검정하기 위한 검정통계량을 구하시오. (반올림하여 소숫점 둘째 자리까지 계산)
- 문제 3: 위의 통계량에 대한 p-값을 구하고 (반올림하여 소숫점 넷째 자리까지 계산), 유의수준 0.05하에서 가설검정의 결과를 채택/기각 중 하나로 선택하시오.
      

## 데이터 불러오기

## 구글 드라이브 연동

In [1]:
from google.colab import drive
drive.mount("/content/drive")

Mounted at /content/drive


In [3]:
import pandas as pd
df = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/2024_2025/imbank/dataset/toothgrowth.csv")
df.head()

len supp  dose
0   4.2   VC   0.5
1  11.5   VC   0.5
2   7.3   VC   0.5
3   5.8   VC   0.5
4   6.4   VC   0.5

## 가설검정
- 독립표본 T-검정은 정규성 검정 뿐만 아니라, 두 그룹이 등분산성을 띠는지 확인하는 작업 필요
- Levene 검정을 사용한 등분산성 검정이 선행되어야 함
- 가설 설정
  + 귀무가설 : 두 그룹의 분산 같다.
  + 대립가설 : 두 그룹의 분산은 같지 않다.

In [8]:
from scipy import stats
stats.levene(df.loc[df['supp'] == "VC", 'len'], df.loc[df['supp'] == "OJ", "len"])

LeveneResult(statistic=1.2135720656945064, pvalue=0.2751764616144053)

- p_value가 0.275 이므로 귀무가설을 채택한다.

## 문제 1
- 표본평균 $\overline{X_{OJ}}$, $\overline{X_{VC}}$를 구하시오.

In [11]:
vc_mean = df.loc[df['supp'] == 'VC', 'len'].mean()
round(vc_mean, 2)

16.96

In [12]:
oj_mean = df.loc[df['supp'] == 'OJ', 'len'].mean()
round(oj_mean, 2)

20.66

## 문제 2
- 위의 가설을 검정하기 위한 검정통계량을 구하시오. (반올림하여 소숫점 둘째 자리까지 계산)

In [14]:
from scipy import stats
t, p = stats.ttest_ind(df.loc[df['supp'] == 'VC', 'len'],
                       df.loc[df['supp'] == 'OJ', 'len'],
                       equal_var=True)

t, p

(-1.91526826869527, 0.06039337122412849)

## 문제 3.
- 위의 통계량에 대한 p-값을 구하고 (반올림하여 소숫점 넷째 자리까지 계산), 유의수준 0.05하에서 가설검정의 결과를 채택/기각 중 하나로 선택하시오.

In [15]:
if p >= 0.05:
  print("귀무가설 채택")
else:
  print("귀무가설 기각 ==> 대립가설 채택")

귀무가설 채택
